### DISCUSSION TITLE
Pipelines - get started sessions

### DISCUSSION THREAD

**Discussion by metazool** [ukceh-rse/community#16] [General]
The proposal here is to run some [Code Clinic](https://github.com/ukceh-rse/community/discussions/15) sessions on how to get started with Reproducible Analytical Pipelines.

* Demystifying your technology choices
* Learning from positive examples
* Preparing your code
* Finding out about infrastructure options

We know there's wide interest, a lot of different "pockets of excellence" (see the internal recording of the Pipelines Knowledge Share workshop) and enthusiastic advocates for packages like [targets for R](https://books.ropensci.org/targets/) and [Luigi for python](https://luigi.readthedocs.io/en/stable/).

As part of this, can we find projects with a strong pipeline / workflow element within the NERC-CEH Github organisation and related orgs, reach out to the maintainers and tag them?

**Comment by metazool** [comment]
Topic coming up in the RSE meetings about reproducible pipelines in a HPC setting, e.g. JASMIN

One stage of a modelling workflow runs distributed, how to collect status from all its nodes into a task-complete part of a graph.

The code is all R - what's out there for `targets` on HPC, or are there `orderly` examples

In this case the model is built with JAGS and this creates a graph of its state. Annual updates, taxonomic updates, how do you reduce load by minimising how much of the model you need to re-run

https://books.ropensci.org/targets/hpc.html

https://www.r-bloggers.com/2012/04/getting-started-with-jags-rjags-and-bayesian-modelling/

https://bookdown.org/steve_midway/BHME/Ch5.html

https://glennklockwood.com/data-intensive/r/parallel-options.html

cc @dylancarbone @francescamancini

**Comment by FrancescaMancini** [comment]
[like]  Francesca Mancini reacted to your message:
________________________________
